### PR TITLE
Rename publish subcommand.

### PR DESCRIPTION
**Pull request by frankban**
It is now called "charm release" to be more in line with snappy semantics.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/125/
Test PASSed.

**Comment by mhilton** [comment]
:+1:

File: cmd/charm/charmcmd/release.go [hunk 1]
```diff
@@ -26,39 +26,39 @@ type publishCommand struct {
 	resources resourceMap
 }
 
-var publishDoc = `
-The publish command publishes a charm or bundle in the charm store.
-Publishing is the action of assigning one channel to a specific charm
+var releaseDoc = `
+The release command publishes a charm or bundle in the charm store.
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/release.go]: I think "to" reads better than "in" here.

File: cmd/charm/charmcmd/release_test.go [hunk 2]
```diff
@@ -179,8 +179,8 @@ var publishDefaultChannelWarnings = []struct {
 	warning: "",
 }}
 
-func (s *publishSuite) TestPublishWithDefaultChannelSuccessWithWarningIfBugsUrlAndHomePageAreNotSet(c *gc.C) {
-	for i, test := range publishDefaultChannelWarnings {
+func (s *releaseSuite) TestReleaseWithDefaultChannelSuccessWithWarningIfBugsUrlAndHomePageAreNotSet(c *gc.C) {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/release_test.go]: s/Url/URL/

:)

**Comment by rogpeppe** [comment]
LGTM with a couple of trivial suggestion, thanks.

File: cmd/charm/charmcmd/release.go (hunk 1 above)
Comment (frankban) [inline cmd/charm/charmcmd/release.go]: Done.

**Comment by frankban** [comment]
Thank you both for the reviews!
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/126/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Comment by ryan-beisner** [comment]
When will this breaking change be communicated, and when will it affect users of these tools who use traditional stable releases via debs?

**Comment by urosj** [comment]
@ryan-beisner usually @marcoceppi makes a release of charm command.

**Pull request merged**